core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -292,10 +330,20 @@ private static void validate(SnapshotId snapshotId) {
      * @param partial                    allow partial snapshots
      * @param userCreateSnapshotListener listener
      */
-    private void beginSnapshot(ClusterState clusterState, final SnapshotsInProgress.Entry snapshot, final boolean partial, final CreateSnapshotListener userCreateSnapshotListener) {
+    private void beginSnapshot(final ClusterState clusterState,
+                               final SnapshotsInProgress.Entry snapshot,
+                               final boolean partial,
+                               final CreateSnapshotListener userCreateSnapshotListener) {
         boolean snapshotCreated = false;
         try {
-            Repository repository = repositoriesService.repository(snapshot.snapshotId().getRepository());
+            // check if the snapshot name already exists in the repository
+            if (snapshotNameExists(snapshot.snapshot())) {
```

Comment:
Do you feel that way because the check only looks at what is in the repository, so it belongs as a repository level method?